### PR TITLE
misc: switch to luajit

### PR DESCRIPTION
**Pull request by GaetanLepage**
It makes sense to use luajit directly as it should be the fastest solution.

### CODE REVIEW

**Comment by GaetanLepage** [comment]
I don't know what is the issue with the CI...

File: flake/default.nix [hunk 1]
```diff
@@ -11,10 +11,5 @@
   perSystem = {pkgs, ...}: {
     formatter = pkgs.alejandra;
     pre-commit.settings.hooks.alejandra.enable = true;
-
-    # Neovim uses lua 5.1 as it is the version which supports JIT
-    _module.args = {
-      lua = pkgs.lua5_1;
```
Comment (teto) [inline flake/default.nix:17]: `lua = pkgs.luajit` is not enough ?

Comment (teto) [reply flake/default.nix:17]: I find it nice to have a high level lua you can override with alternative interpreters. There are many different lua interpreters like luau one might want to test. Also luajit might be available on fewer platforms than puc lua so if one can easily go back to puc lua, it's better.
Btw how would I override `_module.args` from my flake to run sthg like:"
```
_module.args = {
   lua = toto;
}
```
?

Comment (GaetanLepage) [reply flake/default.nix:17]: Ok, I see the point for a user to be able to override the lua.
However, this snippet does not allow for such an external override.
Actually, it was not the case before.

What we could do to let the user override this is to add a `pkgs.luaNeovim` package that we use in the flake.
Maybe then, the user could override what `pkgs.luaNeovim` is on its side, but I am not entirely sure.

Comment (willruggiano) [reply flake/default.nix:17]: Don't make it too complicated. Override the lua input in packages/neovim.nix, same as we do for tree-sitter, treesitter-parsers, etc.

**Comment by GaetanLepage** [comment]
cc @willruggiano this should be good to go know.

**Comment by willruggiano** [comment]
Can you just remove this in favor of overriding the derivation arg like we do for the other build deps?

**Comment by GaetanLepage** [comment]
Like so ?

**Comment by willruggiano** [comment]
Yeah and then override it here; https://github.com/nix-community/neovim-nightly-overlay/blob/7849e018a09688cd71bde887113954b3ce08739c/flake/packages/neovim.nix#L50

**Review by willruggiano — APPROVED** [review]

**Pull request merged**